leshan-server-core/src/main/java/org/eclipse/leshan/server/client/RegistrationService.java

```diff
@@ -39,13 +39,12 @@
     Registration getByEndpoint(String endpoint);
 
     /**
-     * Returns an unmodifiable list of all registrations.
-     * 
-     * @return the registrations.
-     * @deprecated should be replace by an iterator
+     * Returns an iterator over all registrations. There are no guarantees concerning the order in which the
+     * elements are returned.
+     *
+     * @return an <tt>Iterator</tt> over registrations
      */
-    @Deprecated
-    Collection<Registration> getAllRegistrations();
+    Iterator<Registration> getAllRegistrations();
```

Comment:
`RegistrationService` is the API exposed to Leshan users : you can add listener on registration event, access to registrations ...  `RegistrationService` should not be implemented by users, it just help to hide internal API.

`RegistrationStore` is the API needed by Leshan. Users should implement to be able to persist registration using  technology they want. 

About the iterator, It's a good question. IMHO, the `RegistrationService` should expose an API to go through registrations. The previous one was clearly not compatible with a large number of registrations.  So we decide to replace it by **iterator** API. **paging** API is also a possible alternative. Hard to know which are the more suitable approach. **Users feedback about that are welcomed !**

In all case, users are free to implement their own way to browse registrations in their store. This will just be not so easy to access than if we put it in the `RegistrationService` API